desc/objectives/nae_utils.py

```diff
@@ -48,23 +46,31 @@ def calc_zeroth_order_lambda(qsc, desc_eq, N=None):
         Lbasis_sin = FourierSeries(N=desc_eq.N, NFP=nfp, sym="sin")
     else:
         Lbasis_sin = FourierSeries(N=desc_eq.N, NFP=nfp, sym=False)
+    if qsc is not None:
+        phi = qsc.phi
+        dphi = phi[1] - phi[0]
 
-    grid = LinearGrid(M=0, L=0, zeta=phi, NFP=nfp)
-    Ltrans_sin = Transform(grid, Lbasis_sin, build_pinv=True, method="auto")
+        grid = LinearGrid(M=0, L=0, zeta=phi, NFP=nfp)
+        Ltrans_sin = Transform(grid, Lbasis_sin, build_pinv=True, method="auto")
 
-    # from integrating eqn A20 in
-    # Constructing stellarators with quasisymmetry to high order 2019
-    #  Landreman and Sengupta
-    nu_0 = np.cumsum(qsc.B0 / qsc.G0 * qsc.d_l_d_phi - 1) * np.ones_like(phi) * dphi
-    nu_0_n = Ltrans_sin.fit(nu_0)
+        # from integrating eqn A20 in
+        # Constructing stellarators with quasisymmetry to high order 2019
+        #  Landreman and Sengupta
+        nu_0 = np.cumsum(qsc.B0 / qsc.G0 * qsc.d_l_d_phi - 1) * np.ones_like(phi) * dphi
+        nu_0_n = Ltrans_sin.fit(nu_0)
+
+        # lambda = -iota_0 * nu
+        L_0_n = -qsc.iota * nu_0_n
+    else:
+        # we will fix to eq current on-axis lambda, so just set L_0_n to 0
+        # as it will be unused
+        L_0_n = np.zeros(Lbasis_sin.num_modes)
```

Comment:
Is this related to PR #829?

Reply:
no change was made to the zeroth order, I just rearranged stuff to make creating the near-axis constraint without a qsc eq (i..e so fixing to the equilibrium's current behavior) easier